loadgen/prepare-loadgen.js

```diff
@@ -39,9 +39,9 @@ export async function prepareLoadgen(home, deployPowers) {
   /** @type {import('./contract/agent-prepare-loadgen').LoadgenKit | undefined} */
   let loadgenKit = await E(scratch).get(key);
   if (!loadgenKit) {
-    const { bundleSource } = deployPowers;
+    const { bundleSource, publishBundle } = deployPowers;
     const mintFn = path.join(__dirname, 'contract', 'mintHolder.js');
-    const mintBundle = await bundleSource(mintFn);
+    const mintBundle = E.when(bundleSource(mintFn), publishBundle);
```

Comment:
There is a range of commits to agoric-sdk where publishBundle is provided but does not wait for the acknowledgement from the chain before resolving. If this is intended to be run against those versions, there might be a break in continuity. To get around that problem, you can instead depend on the co-versionsed deploy script helpers for installation, which only began using `publishBundle` after it was fixed.

Reply:
> To get around that problem, you can instead depend on the co-versionsed deploy script helpers for installation

Any pointers on how to use that? I may be willing to merge now, and improve this later. Right now I cannot run daily loadperf anymore.

Reply:
Ruh roh. I never merged. https://github.com/Agoric/agoric-sdk/pull/6117 One moment while I rebase.

Reply:
> Identifier 'require' has already been declared

Maybe just try renaming the `const require = ...` definitions in `packages/deploy-script-support/src/extract-proposal.js` and `createBundles.js` to be a different identifier (`req`)?

Reply:
That helped going a little further but still no dice
```agoric: deploy: running /home/node/workspace/agoric/testnet-load-generator/loadgen/loop.js
agoric: deploy: Deploy script will run with standardthings/esm emulation
agoric: (SyntaxError#1)
SyntaxError#1: The requested module 'file:///home/node/workspace/agoric/agoric-sdk/node_modules/@babel/types/lib/index.js' does not provide an export named 'default'

  at Generator.next (<anonymous>)
  at Generator.next (<anonymous>)
  at Generator.next (<anonymous>)
  at Generator.next (<anonymous>)
  at Generator.next (<anonymous>)
  at Generator.next (<anonymous>)
  at Generator.next (<anonymous>)
  at Generator.next (<anonymous>)
  at Object.<anonymous> (packages/deploy-script-support/src/helpers.js:1:1654)
  at Generator.next (<anonymous>)
  at Object.<anonymous> (/home/node/workspace/agoric/testnet-load-generator/loadgen/prepare-loadgen.js:1:721)
  at Generator.next (<anonymous>)
  at Object.<anonymous> (/home/node/workspace/agoric/testnet-load-generator/loadgen/loop.js:1:1312)
  at Generator.next (<anonymous>)
  at tryOnOpen (packages/agoric-cli/src/deploy.js:407:22)
```

I tried changing from RESM -> NESM and it works correctly then. I think it might be time to leave RESM loadgen behind and abandon some backwards compatibility.

Reply:
For anyone stumbling on this thread, I found a way, which I've added to this PR.